README.md

```diff
@@ -28,7 +29,98 @@ For example, Zarr data in this proposed spec might be represented as:
 },
 ```
 
+### Version 1
+
+Metadata structure in JSON. We note, for future possible binary storage, that "version", "gen" and "templates" should
+be considered attributes, and "refs" as the data that ought to dominate the storage size. The previous definition,
+Version 0, is compatible with the "refs" entry, but here we add features. It will also be possible to *expand*
+this new enhanced spec into Version 0 format.
+
+
+```
+{
+  "version": (required, must be equal to) 1,
+  "templates": (optional, zero or more arbitary keys) {
+    "template_name": jinja-str
+  },
+  "gen": (optional, zero or more items) [
+    "key": (required) jinja-str,
+    "url": (required) jinja-str,
+    "offset": (required) jinja-str,
+    "length": (required) jinja-str,
+    "dimensions": (required, one or more arbitrary keys) {
+      "variable_name": (required) 
+        {"start": (optional) int, "stop": (required) int, "step": (optional) int}
+        OR
+        [int, ...]
+    }
+  ],
+  "refs": (optional, zero or more arbiritary keys) {
+    "key_name": (required) str OR [url(jinja-str), offset(int), length(int)]
+  }
+}
+```
+
+Where:
+- `jinja-str` is a string which will be rendered by jinja2 or its non-python equivalent; i.e., it may be
+  a literal string, or may include "{{..}}" annotations, where 
+  - for the values associated with a template_name, the variables are to be passed in reference URL strings that 
+    use this template
+  - for the values within a "gen" object, variables come from the "dimensions" and "templates"
+- the `str` format of a reference value may be
+  - a string starting "base64:", which will be decoded to binary
+  - any other string, interpreted as ascii data
+
+Here is an example 
+
+```json
+{
+    "version": 1,
+    "templates": {
+        "u": "server.domain/path",
+        "f": "{{c}}"
+    },
+    "gen": [
+        {
+            "key": "gen_key{{i}}",
+            "url": "http://{{u}}_{{i}}",
+            "offset": "{(i + 1) * 1000}",
```

Comment:
```suggestion
             "offset": "({{ i }} + 1) * 1000",
```

Reply:
I should be `"{{(i + 1) * 1000}}"` - to evaluate the whole thing within braces as an expression. Your version would evaluate to the string "(0 + 1) * 1000" for i=0.